### PR TITLE
Refactor query compilation.

### PR DESCRIPTION
**Pull request by cristianberneanu**
Simplifies and improves compilation of anonymizing queries.

### CODE REVIEW

File: src/OpenDiffix.Core.Tests/QueryValidator.Tests.fs [hunk 1]
```diff
@@ -2,33 +2,50 @@ module OpenDiffix.Core.QueryValidatorTests
...
-      [
-        { Name = "str_col"; Type = StringType }
-        { Name = "int_col"; Type = IntegerType } // AID column
-        { Name = "float_col"; Type = RealType }
-        { Name = "bool_col"; Type = BooleanType }
-      ]
+let testTables =
+  [
+    {
+      Name = "anon_table"
```
Comment (pdobacz) [inline src/OpenDiffix.Core.Tests/QueryValidator.Tests.fs:8]: you might instead switch the AccessLevel between `Trusted` and `Direct`. Maybe that's more concise.

File: src/OpenDiffix.Core/Analyzer.fs [hunk 2]
```diff
@@ -309,73 +309,63 @@ let private bucketExpand aidColumnsExpression =
...
+  { selectQuery with GroupBy = groupBy; Having = having }
+
+let private compileQuery anonParams (selectQuery: SelectQuery) =
   let rangeColumns = collectRangeColumns anonParams selectQuery.From
   let aidColumnsExpression = makeAidColumnsExpression rangeColumns
 
-  let isAnonymizing =
-    anonParams.AccessLevel <> Direct
-    && aidColumnsExpression |> Expression.unwrapListExpr |> List.isEmpty |> not
+  let directAccess =
```
Comment (pdobacz) [inline src/OpenDiffix.Core/Analyzer.fs:358]: I'm not necessarily a fan of inverting this condition and calling it `directAccess`. That can be true even despite `AccessLevel <> Direct`, so despite being equivalent, might give rise to some confusion.

Also note, that you're still using the `anonymizing` word in `validate/compileAnonymizingQuery`.

File: src/OpenDiffix.Core/Analyzer.fs [hunk 3]
```diff
@@ -309,73 +309,63 @@ let private bucketExpand aidColumnsExpression =
...
+    anonParams.AccessLevel = Direct
+    || aidColumnsExpression |> Expression.unwrapListExpr |> List.isEmpty
 
-  QueryValidator.validateQuery isAnonymizing anonParams.AccessLevel selectQuery
+  QueryValidator.validateStandardQuery selectQuery
```
Comment (pdobacz) [inline src/OpenDiffix.Core/Analyzer.fs:362]: not about your change, but I'm also finding the term `standard query` confusing - even despite already knowing what that means. Even more so, that the validation involved is about "non-standard" (in layman terms) concepts like `DiffixLowCount` aggregators.

How about we rename this globally to `direct query` (it's probably spread across the repo, but not very frequent, should be manageable)? Then it lines up with `direct access` nicely, and better reflects what is really happening (if we're using Diffix, we're using it directly).

**Review by pdobacz — APPROVED** [review]
Feel free to leave the suggested changes up to my follow-up, in the cases where you agree.

File: src/OpenDiffix.Core.Tests/QueryValidator.Tests.fs (hunk 1 above)
Comment (cristianberneanu) [reply src/OpenDiffix.Core.Tests/QueryValidator.Tests.fs:8]: I would prefer to control this from the query statement instead of needing to pass additional arguments to the helper test functions.

File: src/OpenDiffix.Core/Analyzer.fs (hunk 2 above)
Comment (cristianberneanu) [reply src/OpenDiffix.Core/Analyzer.fs:358]: Inverting the condition name simplified a bit the corresponding checks. 
I don't feel strongly about it so I will leave it up to you to revert the change or maybe use a different name like `isStandardQuery`.

File: src/OpenDiffix.Core/Analyzer.fs (hunk 3 above)
Comment (cristianberneanu) [reply src/OpenDiffix.Core/Analyzer.fs:362]: I'm not convinced that fits well with the intent of this function.
Calling this a `direct query` implies that it isn't an `anonymizing query`. This function should execute common SQL validations that apply to both direct and anonymizing queries. Maybe we should have 3 entry points here: standard SQL checks, direct access checks and anonymizing checks? I will leave it up to you to decide.

**Comment by cristianberneanu** [comment]
I merged this as is so I don't further block you from applying your changes.

File: src/OpenDiffix.Core/Analyzer.fs (hunk 3 above)
Comment (pdobacz) [reply src/OpenDiffix.Core/Analyzer.fs:362]: yeah, same issue with `standard query`, we run this even if the query isn't standard. I'll propose something

Comment (cristianberneanu) [reply src/OpenDiffix.Core/Analyzer.fs:362]: In my head, this somewhat makes sense: the standard SQL validations also apply to anonymizing queries.

Comment (pdobacz) [reply src/OpenDiffix.Core/Analyzer.fs:362]: To me, the confusion is in the gist of the name `standard query`, in some of the context it appears in here. Also in the fact that `validateStandardQuery` doesn't do checks relevant to anonymizing query. Realizing the latter made it pretty straightforward to untangle, let me show my refactor in a minute...
